lib/core/crypto/dart_esr/src/signing_request_manager.dart

```diff
@@ -783,3 +789,41 @@ class SigningRequestUtils {
     return ESRConstants.ChainIdLookup[ChainName.RESERVED];
   }
 }
+
+extension HyphaSigningRequestManager on SigningRequestManager {
+  /// Convert the ESR standard's "chain_id" to our Network
+  /// Both signify a unique identifier for a chain
+  /// But we support fewer chains than ESR does.
+  /// Networks maps to remote config which contains the server node URLs we need
+  /// to use in order to access supported chains.
+  /// Returns: Network
+  /// throws: unsupported network when the chainID can't be parsed.
+  ///
+  static Networks resolveNetwork(List<dynamic> chainId) {
+    if (chainId[0] == 'chain_alias') {
+      // chain_alias officially only supports EOS mainnet, and Telos mainnet, as 1, and 2.
+      if (chainId[1] == 1) {
+        return Networks.eos;
+      } else if (chainId[1] == 2) {
+        return Networks.telos;
+      } else {
+        throw 'unsupported network alias ${chainId[1]}';
+      }
+    } else if (chainId[0] == 'chain_id') {
```

Comment:
same here. Not sure how to improve it tho.

Reply:
so it's a list and the first element in the list describes whether it's an ID or an alias. 

The alias is even worse, it's sort of a list of constants they're indexing...